### PR TITLE
[RLlib] Contrib Folder Restructuring

### PR DESCRIPTION
**Pull request by michaelzhiluo**
Moved `ray.rllib.contrib` folders to `ray.rllib.contrib.agents` and `ray.rllib.contrib.env` respectively. It still maintains backwards compatibility with `ray.rllib.contrib.alpha_zero` or other agents/envs with deprecation warnings.

## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [x] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [x] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [x] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: rllib/contrib/__init__.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
+import sys
+import ray.rllib.contrib.agents.alpha_zero as AlphaZero
+import ray.rllib.contrib.agents.bandits as Bandits
+import ray.rllib.contrib.agents.maddpg as Maddpg
+import ray.rllib.contrib.agents.random_agent as RandomAgent
+import ray.rllib.contrib.env.sumo as Sumo
+from ray.rllib.utils.deprecation import deprecation_warning
+
+# Will be deprecated soon, old path names will still give the right class
```
Comment (sven1977) [inline rllib/contrib/__init__.py:9]: Great solution!

**Review by sven1977 — APPROVED** [review]
Thanks for the cleanup @michaelzhiluo !

**Comment by michaelzhiluo** [comment]
@sven1977 Should be good to go, linter errors are resolved

**Comment by sven1977** [comment]
@michaelzhiluo Could you fix the LINTer and pull from master once more?
Thanks :)

**Comment by sven1977** [comment]
Let's get this merged @michaelzhiluo :) Just re-merged with master and waiting for testing to complete ...

**Comment by bveeramani** [comment]
## :bangbang: ACTION REQUIRED :bangbang:

We've switched our code formatter from YAPF to Black (see #21311).

To prevent issues with merging your code, here's what you'll need to do:

1. Install Black
```
pip install -I black==21.12b0
```

2. Format changed files with Black
```
curl -o format-changed.sh https://gist.githubusercontent.com/bveeramani/42ef0e9e387b755a8a735b084af976f2/raw/7631276790765d555c423b8db2b679fd957b984a/format-changed.sh
chmod +x ./format-changed.sh
./format-changed.sh
rm format-changed.sh
```

3. Commit your changes.
```
git add --all
git commit -m "Format Python code with Black"
```

4. Merge master into your branch.
```
git pull upstream master
```

5. Resolve merge conflicts (if necessary).

After running these steps, you'll have the updated `format.sh`.

**Pull request closed without merging**